app/src/main/java/org/thoughtcrime/securesms/dependencies/SessionUtilModule.kt

```diff
@@ -33,4 +35,9 @@ object SessionUtilModule {
             registerListener(context as ConfigFactoryUpdateListener)
         }
 
+    @Provides
+    fun provideMnemonicLoader(@ApplicationContext context: Context) = MnemonicLoader(context)
+
+    @Provides
+    fun provideMnemonicCodec(loader: MnemonicLoader) = MnemonicCodec(loader::loadFileContents)
```

Comment:
same here, and should they be singletons?

Reply:
might be nice to just pass `loader` in, as the `String -> String` is not necessarily 100% clear.